sql/wsrep_binlog.cc

```diff
@@ -380,6 +381,48 @@ static int wsrep_write_cache_inc(wsrep_t*  const wsrep,
     return err;
 }
 
+static int prepend_binlog_control_event(wsrep_t* const wsrep, THD* const thd)
+{
+    if ((thd->variables.option_bits & OPTION_BIN_LOG) != 0)
+      return 0;
+
+    IO_CACHE tmp_io_cache;
+    if (open_cached_file(&tmp_io_cache, mysql_tmpdir, TEMP_PREFIX,
```

Comment:
I think we must fill the cache with zeros to be on the safer side. 
```suggestion
    IO_CACHE tmp_io_cache;
    memset(&tmp_io_cache, 0, sizeof(stmt_cache_log));
    if (open_cached_file(&tmp_io_cache, mysql_tmpdir, TEMP_PREFIX,
```
The same has been done in `THD::binlog_setup_trx_data()` as well.

Reply:
I don't think so. Please check `wsrep_to_buf_helper`.

Reply:
OK, after the discussion, let's initialize it in both places :)